RHDEVDOCS 6346 configure PAC pipeline run cancel-in-progress, plus mo… (#88596)
openshift/openshift-docs: modules/op-parameters-pipeline-run-using-pipelines-as-code.adoc

// This module is included in the following assemblies:
// * pac/managing-pipeline-runs-pac.adoc

:_mod-docs-content-type: REFERENCE
[id="parameters-pipeline-run-using-pipelines-as-code_{context}"]
= Parameters in a pipeline run specification

> You can use parameters in a pipeline run specification to provide information related to the commit to which the pipeline run is related and to use the temporary GitHub App token for Github API operations.

Reviewer: ```suggestion
You can use parameters in a pipeline run specification to provide information about the commit related to the pipeline run and to use the temporary GitHub App token for Github API operations.
```

[id="commits-parameters-pipeline-run-using-pipelines-as-code_{context}"]
== Commit and URL information

You can specify the parameters of your commit and URL by using dynamic, expandable variables with the `{{<var>}}` format. Currently, you can use the following variables:

* `{{repo_owner}}`: The repository owner.
* `{{repo_name}}`: The repository name.
* `{{repo_url}}`: The repository full URL.
* `{{revision}}`: Full SHA revision of a commit.
* `{{sender}}`: The username or account ID of the sender of the commit.
* `{{source_branch}}`: The branch name where the event originated.
* `{{target_branch}}`: The branch name that the event targets. For push events, it is the same as the `source_branch`.
* `{{pull_request_number}}`: The pull or merge request number, defined only for a `pull_request` event type.
* `{{git_auth_secret}}`: The secret name that is generated automatically with the Git provider token for checking out private repos.

[id="token-parameters-pipeline-run-using-pipelines-as-code_{context}"]

> == Temporary GitHub App token for Github API operations

Reviewer: ```suggestion
== Temporary GitHub App token for GitHub API operations
```

You can use the temporary installation token generated by {pac} from the GitHub App to access the GitHub API. The GitHub App generates a key for private repositories in the `git-provider-token` key. You can use the `{{git_auth_secret}}` dynamic variable in pipeline runs to access this key.

> For example, to add a comment to a pull request, you can use the `github-add-comment` task from {tekton-hub} by using a {pac} annotation:

Reviewer: ```suggestion
For example, to add a comment to a pull request, you can use a {pac} annotation to execute the `github-add-comment` task from {tekton-hub}:
```
It's actually not clear to me how this is an example of the above / I'm not seeing the relationship between the token/key and the `github-add-comment` task. Are we saying: After you gain access to the GitHub API, you can use a {pac} annotation to perform GitHub actions, such as comment on a pull request with the `github-add-comment` task? 

If so, it might be a good idea to add some info/language to close that gap.

Author: I do see where the confusioon happened and I'm putting in a fix for that.

[source,yaml]
----
...
  pipelinesascode.tekton.dev/task: "github-add-comment"
...
----

> You can then add a task to the `tasks` section or `finally` tasks in the pipeline run definition:

Reviewer: It sounds like this is a sequential step, which sounds like a procedure. Even if it is optional, if it must be done after updating the yaml file, then maybe this could be formatted as a proc module.

Author: These are not really sequential steps, and this is pre-existing reference information. I do see how this looked like sequential steps and so I reworked it into a single example. It also made the example compliant with the "complete YAML" requirement.

// This module is included in the following assemblies:
// * pac/managing-pipeline-runs-pac.adoc

:_mod-docs-content-type: REFERENCE

[id="parameters-pipeline-run-using-pipelines-as-code_{context}"]
= Using parameters in a pipeline run specification

You can use parameters in a pipeline run specification to provide information related to the commit to which the pipeline run is related and to use the temporary GitHub App token for Github API operations.

[id="commits-parameters-pipeline-run-using-pipelines-as-code_{context}"]
== Parameterizing commits and URLs

You can specify the parameters of your commit and URL by using dynamic, expandable variables with the `{{<var>}}` format. Currently, you can use the following variables:

* `{{repo_owner}}`: The repository owner.
* `{{repo_name}}`: The repository name.
* `{{repo_url}}`: The repository full URL.
* `{{revision}}`: Full SHA revision of a commit.

> * `{{sender}}`: The username or account id of the sender of the commit.

Reviewer 2: ```suggestion
* `{{sender}}`: The username or account ID of the sender of the commit.
```
per IBM Style

* `{{source_branch}}`: The branch name where the event originated.

> * `{{target_branch}}`: The branch name that the event targets. For push events, it's the same as the `source_branch`.

Reviewer 2: ```suggestion
* `{{target_branch}}`: The branch name that the event targets. For push events, it is the same as the `source_branch`.
```
Elsewhere, you have "it is" so I would just be consistent.

* `{{pull_request_number}}`: The pull or merge request number, defined only for a `pull_request` event type.

> * `{{git_auth_secret}}`: The secret name that is generated automatically with Git provider's token for checking out private repos.

Reviewer 2: ```suggestion
* `{{git_auth_secret}}`: The secret name that is generated automatically with the Git provider's token for checking out private repos.
```
